### ISSUE TITLE
Using both namePrefix and namespace changes the metadata.name in the namespace

### ISSUE THREAD

**Issue by k8sz** [kubernetes-sigs/kustomize#2371]
kustomization.yaml
```yaml
namePrefix: prod-
```
namespace.yaml
```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: prod
```
kustomize build

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: prod-prod
```

**Comment by k8sz** [comment]
This is not what I wanted.
The namespace I need is prod.

**Comment by gmolkvk** [comment]
Maybe you should look into the meaning of the word "prefix"

**Comment by k8sz** [comment]
I just don't want this prefix to appear in kind: Namespace

**Comment by ageekymonk** [comment]
Most of the time we create resources within a namespace. Having to skip the prefix for namespace would be immensely helpful as we can create all the resources within a namespace (including the namespace) in one go. Without this we are now creating namespace separately which i think is not ideal. 

If this is not acceptable for some reason, it would be good to have a configuration for prefix to exclude certain kind (in our case the namespace). This way we can achieve the same result. 

My two cents.

**Comment by ne1000** [comment]
@k8sz I had met the same issues too,  did you fixed it ?

**Comment by ageekymonk** [comment]
@ne1000 yes. It is fixed now. It got merged #2742

**Comment by ne1000** [comment]
@ageekymonk good, thanks.
